src/schema/schema_create.c

```diff
@@ -433,6 +433,57 @@ __wt_find_import_metadata(WT_SESSION_IMPL *session, const char *uri, const char
     return (0);
 }
 
+/*
+ * __schema_is_tiered_storage_shared --
+ *     Check whether the table is configured for a tiered storage shared.
+ */
+static void
+__schema_is_tiered_storage_shared(
+  WT_SESSION_IMPL *session, const char *config, bool *is_tiered_storage_shared)
+{
+    WT_CONFIG_ITEM cval;
+
+    *is_tiered_storage_shared = true;
+
+    if (__wt_config_getones(session, config, "source", &cval) == 0 && cval.len != 0)
+        *is_tiered_storage_shared = false;
+    else if (__wt_config_getones(session, config, "type", &cval) == 0 &&
+      !WT_STRING_MATCH("file", cval.str, cval.len))
+        *is_tiered_storage_shared = false;
+    else if ((S2C(session)->bstorage == NULL) ||
+      (__wt_config_getones(session, config, "tiered_storage.name", &cval) == 0 && cval.len != 0 &&
+        WT_STRING_MATCH("none", cval.str, cval.len)))
+        *is_tiered_storage_shared = false;
+    else if (!S2C(session)->bstorage->tiered_shared ||
+      ((__wt_config_getones(session, config, "tiered_storage.shared", &cval) == 0) && !cval.val))
+        *is_tiered_storage_shared = false;
+}
+
+/*
+ * __schema_tiered_shared_colgroup_source --
+ *     Get the tiered storage shared URI of the data source for a column group.
+ */
+static int
+__schema_tiered_shared_colgroup_source(
+  WT_SESSION_IMPL *session, WT_TABLE *table, bool active, WT_ITEM *buf)
+{
+    size_t len;
+    const char *prefix, *suffix, *tablename;
+
+    tablename = table->iface.name + strlen("table:");
+    if (active) {
+        prefix = "file";
+        len = strlen(prefix);
+        suffix = ".wt";
+    } else {
+        prefix = "tiered";
+        len = strlen(prefix);
+        suffix = "";
```

Comment:
Hmm, this is surprising. I was expecting that if active, the suffix would be `.active` and if not active the suffix would be `.shared`.

Also `len = strlen(prefix)` could be pulled out since it is identical for both cases.

Reply:
This function returns the underlying source for the column group. There is another function that returns column group names based on the `active` flag. Fixed the length part.

Reply:
Thanks. Can you expand the comment to clearly state the naming of the components of the shared table? I.e. For a shared tiered table named `table:name` the active table is always `file:name.wt` and the shared table is `tiered:name` which points to the shared components.

Reply:
Added the suggested comment to explain the source for active and shared components.